src/OrchardCore.Modules/OrchardCore.Tenants/Services/TenantValidator.cs

```diff
@@ -84,12 +84,20 @@ public async Task<IEnumerable<ModelError>> ValidateAsync(TenantViewModel model)
 
             var allOtherShells = allSettings.Where(t => !String.Equals(t.Name, model.Name, StringComparison.OrdinalIgnoreCase));
 
-            if (allOtherShells.Any(tenant => String.Equals(tenant.RequestUrlPrefix, model.RequestUrlPrefix?.Trim(), StringComparison.OrdinalIgnoreCase) && (tenant.RequestUrlHost?.Contains(model.RequestUrlHost, StringComparison.OrdinalIgnoreCase) ?? false)))
+            if (allOtherShells.Any(tenant => String.Equals(tenant.RequestUrlPrefix, model.RequestUrlPrefix?.Trim(), StringComparison.OrdinalIgnoreCase) && IsUrlHostExist(tenant.RequestUrlHost, model.RequestUrlHost)))
```

Comment:
Please thoroughly test before you request a review. It has happened multiple times that I just try to break validation in some simple way, and I succeed. This doesn't cause a validation error now:

1. Create tenant with the host "example1.com".
2. Create tenant with the host "example1.com, example2.com".
3. You don't get a validation error.

Reply:
I just test the both cases before I request your review, seems you bring a new case ;) If you remember the first case the previous tenant contains the new one, which is what the PR is started for, but this case is another one. So, I should test that the new tenant hosts exists

Reply:
I think it's better to remove the `Conains()` because a complex case may be:

1. Create tenant with the host "example1.com, example2.com".
2. Create tenant with the host "example3.com, example2.com, example4.com".
3. You get a validation error.

Reply:
The order in what you create the tenants, of course, shouldn't matter. How you implement it is up to you.

Reply:
Seems the issue from the beginning because I stick with using `Contains()` and solve the issue from on angle, the latest changes should fix the issue permenantly

I need to do the last test from OC admin

Reply:
Also add a unit test.